### PR TITLE
2335: Avoid creating unneeded objects

### PR DESCRIPTION
**Pull request by pavelrappo**
This PR removes needless objects, some of which are resources. Removing those objects should have no effect.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Issue
 * [SKARA-2335](https://bugs.openjdk.org/browse/SKARA-2335): Avoid creating unneeded objects (**Bug** - P5)

### Reviewers
 * [Zhao Song](https://openjdk.org/census#zsong) (@zhaosongzs - **Reviewer**) ⚠️ Review applies to [4c16b10b](https://git.openjdk.org/skara/pull/1680/files/4c16b10b647dfdde961daf86107bf91fed4cc957)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/skara.git pull/1680/head:pull/1680` \
`$ git checkout pull/1680`

Update a local copy of the PR: \
`$ git checkout pull/1680` \
`$ git pull https://git.openjdk.org/skara.git pull/1680/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1680`

View PR using the GUI difftool: \
`$ git pr show -t 1680`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/skara/pull/1680.diff">https://git.openjdk.org/skara/pull/1680.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/skara/pull/1680#issuecomment-2239052022)

### CODE REVIEW

File: vcs/src/test/java/org/openjdk/skara/vcs/RepositoryTests.java [hunk 1]
```diff
@@ -742,7 +742,6 @@ void testNonCheckedOutRepositoryIsHealthy(VCS vcs) throws IOException {
     void testBranchesOnEmptyRepository(VCS vcs) throws IOException {
         try (var dir = new TemporaryDirectory()) {
             var r = TestableRepository.init(dir.path(), vcs);
-            var expected = vcs == VCS.GIT ? List.of() : List.of(new Branch("default"));
             assertEquals(List.of(), r.branches());
```
Comment (zhaosongzs) [inline vcs/src/test/java/org/openjdk/skara/vcs/RepositoryTests.java:745]: I am just curious about the expected behavior when testing on a mercurial (hg) repository.

When I initialize a new hg repository and run `hg branch`, it returns "default". However, when I use `hg branches`, it returns empty. So is "default" branch ignored?

**Review by zhaosongzs — APPROVED** [review]
Looks good

File: vcs/src/test/java/org/openjdk/skara/vcs/RepositoryTests.java (hunk 1 above)
Comment (pavelrappo) [reply vcs/src/test/java/org/openjdk/skara/vcs/RepositoryTests.java:745]: From my experiments, Git (2.45.2) and Mercurial (6.8) behave identically in that regard. Here's a newly initialised Git repo:

    % git branch --list --all
    % git branch --show-current
    master
    %

It's only after I make the first commit to the default branch, be it Git or Mercurial, that branch shows up in the output.

**Pull request closed without merging**